check.py fails on macOS

Running `check.py` on macOS fails on gcc torture tests.

```
build:  /usr/bin/gcc /Users/pepyakin/dev/etc/binaryen/test/example/c-api-hello-world.c -c -o example.o -I/Users/pepyakin/dev/etc/binaryen/src -g -Lbin/../lib -pthread
clang: warning: argument unused during compilation: '-Lbin/../lib' [-Wunused-command-line-argument]
   c-api-hello-world.c /Users/pepyakin/dev/etc/binaryen/test/example/c-api-hello-world.c /Users/pepyakin/dev/etc/binaryen/test/example/c-api-hello-world.txt
link:  /usr/bin/g++ -std=c++11 example.o -lbinaryen -I/Users/pepyakin/dev/etc/binaryen/src -g -lasmjs -lsupport -Lbin/../lib -pthread -o bin/example -Wl,-rpath=$ORIGIN/../lib
clang: warning: argument unused during compilation: '-pthread' [-Wunused-command-line-argument]
ld: unknown option: -rpath=$ORIGIN/../lib
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Traceback (most recent call last):
  File "./check.py", line 740, in <module>
    run_gcc_torture_tests()
  File "./check.py", line 584, in run_gcc_torture_tests
    os.remove(output_file)
OSError: [Errno 2] No such file or directory: 'bin/example'
```

Seems like it happens because `clang` pretends to be `gcc`.

```
gcc --version                                                                                                                                                                           
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

I tried solving it by installing gcc via `brew install gcc` and using `CC=gcc-8 CXX=g++-8 ./check.py` however I still receive the same error:

<details>
<summary>Error details</summary>
<pre>
build:  gcc-8 /Users/jayphelps/Projects/jayphelps/binaryen/test/example/c-api-hello-world.c -c -o example.o -I/Users/jayphelps/Projects/jayphelps/binaryen/src -g -Lbin/../lib -pthread
   c-api-hello-world.c /Users/jayphelps/Projects/jayphelps/binaryen/test/example/c-api-hello-world.c /Users/jayphelps/Projects/jayphelps/binaryen/test/example/c-api-hello-world.txt
link:  g++-8 -std=c++11 example.o -Lbin/../lib -lbinaryen -I/Users/jayphelps/Projects/jayphelps/binaryen/src -g -pthread -o bin/example -Wl,-rpath=$ORIGIN/../lib
ld: unknown option: -rpath=$ORIGIN/../lib
collect2: error: ld returned 1 exit status
Traceback (most recent call last):
  File "./check.py", line 675, in <module>
    sys.exit(main())
  File "./check.py", line 657, in main
    run_gcc_tests()
  File "./check.py", line 566, in run_gcc_tests
    os.remove(output_file)
OSError: [Errno 2] No such file or directory: 'bin/example'
</pre>
</details>
--

I noted `ld: unknown option: -rpath=$ORIGIN/../lib` and some googling suggested it might need to be a comma instead `-rpath,$ORIGIN/../lib` so I change that line in `check.py`.

I also noticed that an old PR had the [rpath stuff excluded on Mac](https://github.com/WebAssembly/binaryen/pull/517/files#diff-80ee07c6b10902b1bae15660a736de65R615) though that change was later lost. I'm not sure what's right.

That resolved the error, but then I got `fatal error: _stdio.h: No such file or directory`

More googling suggested that I needed to update my xcode "command line tools" so I used:

```
sudo softwareupdate -i "Command Line Tools (macOS High Sierra version 10.13) for Xcode-9.4"
```

PHEW. Now it works!

Aside from the `-rpath,$ORIGIN/../lib` change, I wonder if there's anything that can be done to make this easier for Mac users, or if we just need to provide some additional instructions for setup.

--- a/auto_update_tests.py
+++ b/auto_update_tests.py
@@ -225,7 +225,7 @@ def update_example_tests():
     print os.getcwd()
     subprocess.check_call(extra)
     # Link against the binaryen C library DSO, using rpath
-    cmd = ['example.o', '-L' + libdir, '-lbinaryen', '-Wl,-rpath=' + os.path.abspath(libdir)] + cmd
+    cmd = ['example.o', '-L' + libdir, '-lbinaryen', '-Wl,-rpath,' + os.path.abspath(libdir)] + cmd
     print '  ', t, src, expected
     if os.environ.get('COMPILER_FLAGS'):
       for f in os.environ.get('COMPILER_FLAGS').split(' '):

--- a/check.py
+++ b/check.py
@@ -547,7 +547,7 @@ def run_gcc_tests():
         print 'build: ', ' '.join(extra)
         subprocess.check_call(extra)
         # Link against the binaryen C library DSO, using an executable-relative rpath
-        cmd = ['example.o', '-L' + os.path.join(options.binaryen_bin, '..', 'lib'), '-lbinaryen'] + cmd + ['-Wl,-rpath=$ORIGIN/../lib']
+        cmd = ['example.o', '-L' + os.path.join(options.binaryen_bin, '..', 'lib'), '-lbinaryen'] + cmd + ['-Wl,-rpath,$ORIGIN/../lib']
       else:
         continue
       print '  ', t, src, expected

